cumulus/client/chain-spec-extension/Cargo.toml

```diff
@@ -0,0 +1,14 @@
+[package]
+name = "cumulus-client-chain-spec-extension"
```

Comment:
Why do we need a new crate for them?

Reply:
My idea was that we can refer to this crate from small tools like chain-spec-builder without pulling in the whole world of dependencies. Which would be the case if I put it in some other bigger crate.

Reply:
Could be also done with features, one of which could be 'api' exposing necessary types only.

Reply:
or maybe move it to sc_chain_spec::common, as it is typical/common of parachains?

Reply:
https://github.com/paritytech/polkadot-sdk/pull/4685#discussion_r1651521447 given this, we should revert these changes. Merge the pr with the `codeSubstitute` support and add a new one that ignores the unknown fields.

Reply:
Tracking of follow up https://github.com/paritytech/polkadot-sdk/issues/4873